tests/test_zero_data_parallel/common.py

```diff
@@ -71,11 +71,11 @@ def check_grads_padding(model, zero_model, loose=False):
         chunks = torch.flatten(p.grad).chunk(dist.get_world_size())
         if rank >= len(chunks):
             continue
-        grad = chunks[rank]
+        grad = chunks[rank].float()
```

Comment:
why convert to float?
When I see the name check_grads_padding, I won't expect you cast my model to fp16 during the check.

Reply:
The original model is fp16, so the grad is also fp16.

Reply:
what is original model?
grad = chunks[rank].float() if zero_grad.dtype == torch.float else chunks[rank]

Reply:
In `test_shard_model_v2.py`, we call `model.half()`. So the parameter and gradient of the model which our `ShardedModel` compares are always fp16. And as we store fp32 grad shard for `ShardedModel`, `zero_grad` is always fp32 here.